### ISSUE TITLE
MongoItemReaderBuilder shouldn't be throwing an error related to the limit in query when pageSize is provided

### ISSUE THREAD

**Issue by kexsteve** [spring-projects/spring-batch#3673]
https://github.com/spring-projects/spring-batch/blob/8e8f9c8101b52b64050d8e92363de2d28319e3b0/spring-batch-infrastructure/src/main/java/org/springframework/batch/item/data/builder/MongoItemReaderBuilder.java#L290

For example: 
```
new MongoItemReaderBuilder<Document>()
            .collection("test")
            .template(mongoTemplate)
            .targetType(Document.class)
            .query(Query.query(Criteria.where("active").is(false))
            .sorts(ImmutableMap.of("_id", Sort.Direction.ASC))
            .pageSize(100)
            .build();
```

The line throw an error in MongoItemReaderBuilder:
Assert.state(this.query.getLimit() != 0, "PageSize in Query object was ignored.")

This has been just a warning in MongoItemReader:
log.warn("PageSize in Query object was ignored. Please set it by MongoItemReader.setPageSize().");

**Comment by benas** [comment]
I do confirm, this is indeed inconsistent with the `MongoItemReader`. However, the issue is in the reader, before being in the builder. When I set a query limit without a page size, I have the warning:

```java
@Test
public void testMongoItemReaderWithQueryLimit() throws Exception {
	MongoItemReader<String> mongoItemReader = new MongoItemReader<>();
	mongoItemReader.setName("mongoReader");
	mongoItemReader.setCollection("test");
	mongoItemReader.setTemplate(template);
	mongoItemReader.setTargetType(String.class);
	mongoItemReader.setQuery(Query.query(Criteria.where("active").is(false)).limit(1000));
	mongoItemReader.setSort(new HashMap<String, Sort.Direction>() {{ put("_id", Sort.Direction.ASC); }});
	mongoItemReader.afterPropertiesSet();

	Assertions.assertNotNull(mongoItemReader);
}
```

This test passes, and I see the following warning:

```
[main] WARN org.springframework.batch.item.data.MongoItemReader - PageSize in Query object was ignored. Please set it by MongoItemReader.setPageSize()
```

However, even if I set the page size as mentioned in the warning, I still see the same warning..

```java
@Test
public void testMongoItemReaderWithQueryLimit() throws Exception {
	MongoItemReader<String> mongoItemReader = new MongoItemReader<>();
	mongoItemReader.setName("mongoReader");
	mongoItemReader.setCollection("test");
	mongoItemReader.setTemplate(template);
	mongoItemReader.setTargetType(String.class);
	mongoItemReader.setQuery(Query.query(Criteria.where("active").is(false)).limit(1000));
	mongoItemReader.setSort(new HashMap<String, Sort.Direction>() {{ put("_id", Sort.Direction.ASC); }});
	mongoItemReader.setPageSize(100);
	mongoItemReader.afterPropertiesSet();

	Assertions.assertNotNull(mongoItemReader);
}
```

This is because the condition on the query [here](https://github.com/spring-projects/spring-batch/blob/e0f78d1504b0d94cbdea4e1739f2279e0843d9fb/spring-batch-infrastructure/src/main/java/org/springframework/batch/item/data/MongoItemReader.java#L245-L247) is incorrect as it does not involve the page size. (and btw, the message `PageSize in Query object was ignored` is incorrect as well, there is no `PageSize` parameter in the `Query` type).

In hindsight, this warning is not needed IMO, because the page size defaults to 10, and could be set by the user to a non-zero value. So in the end, regardless of the query limit, the page size is set. I think we need to fix the reader first, otherwise we would make the builder consistent with an incorrect reader.

**Comment by parikshitdutta** [comment]
Hi @benas can you please take a look at my PR #3691 for resolving this issue? or please assign it to the respective group to review.

Thank you.

**Comment by benas** [comment]
@parikshitdutta Thank you for the PR! I will take a look.

**Comment by parikshitdutta** [comment]
thanks @benas, pleasure is mine! looking forward to take some more issues across open spring framework issues, while we fight covid!

**Comment by parikshitdutta** [comment]
Hi @benas, if you envision any change for this PR that can help, please let me know.

Thank you.

**Comment by benas** [comment]
Resolved with 56fbc270e22a3f28a188307a32fe5b85264ddec3 and backported to 4.2.x.
